src/cluster_migrateslots.c

```diff
@@ -353,6 +358,138 @@ void fireModuleSlotMigrationEvent(slotMigrationJob *job, int subevent) {
+        sdsfreesplitres(slot_range_strs, num_slot_ranges);
+        return NULL;
+    }
+    list *slot_ranges = createSlotRangeList();
+    for (int j = 0; j < num_slot_ranges; j++) {
+        sds slot_range_str = slot_range_strs[j];
+        slotRange *slot_range = decodeSlotRange(slot_range_str);
+        if (slot_range == NULL) {
+            sdsfreesplitres(slot_range_strs, num_slot_ranges);
+            listRelease(slot_ranges);
+            return NULL;
+        }
+        listAddNodeTail(slot_ranges, slot_range);
+    }
+    sdsfreesplitres(slot_range_strs, num_slot_ranges);
+    return slot_ranges;
+}
+
+int clusterDecodeSlotImportsAuxField(int rdbflags, sds s) {
+    if (!server.cluster_enabled || s == NULL) return C_OK;
+
+    if ((rdbflags & RDBFLAGS_AOF_PREAMBLE) != 0) return C_OK;
+
+    list *new_slot_migrations = listCreate();
+    listSetFreeMethod(new_slot_migrations, freeSlotMigrationJob);
+
+    int num_fields = 0;
+    sds *import_fields = NULL;
+    int num_imports = 0;
+    sds *import_strs = sdssplitlen(s, sdslen(s), "/", 1, &num_imports);
+    for (int i = 0; i < num_imports; i++) {
+        import_fields = sdssplitlen(import_strs[i], sdslen(import_strs[i]), ":", 1, &num_fields);
+
+        if (num_fields < 3) goto error;
+        sds job_name = import_fields[0];
+        sds node_name = import_fields[1];
+        sds slot_range_list_str = import_fields[2];
+
+        /* Decode the slot ranges */
+        list *slot_ranges = decodeSlotRangeList(slot_range_list_str);
+        if (!slot_ranges) goto error;
+
+        /* Try to find the corresponding node, or else just insert with no human name. */
+        char *source_node_human_name = NULL;
+        clusterNode *source_node = clusterLookupNode(node_name, CLUSTER_NAMELEN);
+        if (source_node) source_node_human_name = source_node->human_nodename;
+
+        slotMigrationJob *new_import = createSlotImportJob(NULL, node_name, source_node_human_name, job_name, slot_ranges);
+        listAddNodeTail(new_slot_migrations, new_import);
+        sdsfreesplitres(import_fields, num_fields);
+        import_fields = NULL;
+    }
+
+    listIter li;
+    listNode *ln;
+    listRewind(new_slot_migrations, &li);
+    while ((ln = listNext(&li))) {
+        slotMigrationJob *job = ln->value;
+        listAddNodeHead(server.cluster->slot_migration_jobs, job);
+        serverLog(LL_NOTICE,
+                  "Slot import discovered during snapshot load: %s",
+                  job->description);
+        fireModuleSlotMigrationEvent(job, VALKEYMODULE_SUBEVENT_ATOMIC_SLOT_MIGRATION_IMPORT_STARTED);
+    }
+
+    listSetFreeMethod(new_slot_migrations, NULL);
+    listRelease(new_slot_migrations);
+    sdsfreesplitres(import_strs, num_imports);
+    return C_OK;
+
+error:
+    listRelease(new_slot_migrations);
+    sdsfreesplitres(import_strs, num_imports);
+    sdsfreesplitres(import_fields, num_fields);
+    return C_ERR;
+}
+
+int clusterRegisterSlotImportsAuxFields(void) {
```

Comment:
Without mentioning "RDB" in the name, it's easy to think this is about Cluster aux fields (something we store in nodes.conf and correspond to ping extensions).

```suggestion
int clusterRegisterSlotImportsRdbAuxFields(void) {
```

Reply:
Done